llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -610,8 +610,11 @@ static Value *foldLogOpOfMaskedICmps(Value *LHS, Value *RHS, bool IsAnd,
       APInt NewMask = *ConstB & *ConstD;
       if (NewMask == *ConstB)
         return LHS;
```

Comment:
why is not the same check needed for LHS?

Reply:
If LHS is poison, the result of logical and/or is also poison.